### PR TITLE
Lang switch

### PR DESCRIPTION
**Pull request by zubcevic**
Added functionality to switch from the browser default language to another language. And added dutch and German intro welcome pages.

### CODE REVIEW

File: src/main/java/org/owasp/webgoat/container/MvcConfiguration.java [hunk 1]
```diff
@@ -196,6 +199,25 @@ public Language language(LocaleResolver localeResolver) {
         return new Language(localeResolver);
     }
 
+    @Bean
+    public LocaleResolver localeResolver() {
+        SessionLocaleResolver localeResolver = new SessionLocaleResolver();
+        //localeResolver.setDefaultLocale(Locale.ENGLISH);
```
Comment (nbaars) [inline src/main/java/org/owasp/webgoat/container/MvcConfiguration.java:205]: Default is English, we can remove the comment?

File: src/main/java/org/owasp/webgoat/container/MvcConfiguration.java [hunk 2]
```diff
@@ -205,10 +227,10 @@ public Messages messageSource(Language language) {
         return messages;
     }
 
-    @Bean
+    /*@Bean
```
Comment (nbaars) [inline src/main/java/org/owasp/webgoat/container/MvcConfiguration.java:230]: Can be removed?

File: src/main/resources/lessons/webgoat_introduction/documentation/Introduction.adoc [hunk 3]
```diff
@@ -1,6 +1,12 @@
 == What is WebGoat?
 ---
 
+ifeval::["{lang}" == "de"]
+:caution-caption: Achtung
+test
```
Comment (nbaars) [inline src/main/resources/lessons/webgoat_introduction/documentation/Introduction.adoc:6]: Think this is for test?

File: src/main/resources/lessons/webgoat_introduction/documentation/Introduction_nl.adoc [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+
+Voel je vrij om WebGoat eens flink uit te testen en je kennis en ervaring op dit gebied te verbeteren.
+
+Je moet met een hacking mindset proberen de WebGoat applicatie oefeningen op te lossen.
+
+Bedankt voor je interesse!
+
+*Het team van WebGoat*
+
+#De meeste tekst is in het Engels. U kunt van taal wisselen via de dropdown in het menu. U bent ook van harte uitgenodigd om teksten te vertalen en WebGoat beter te maken.#
```
Comment (nbaars) [inline src/main/resources/lessons/webgoat_introduction/documentation/Introduction_nl.adoc:18]: en WebGoat beter te maken --> om WebGoat te verbeteren

**Review by nbaars — COMMENTED** [review]
Nice change. I'm not against having the option to support multiple language, but we do deliver a lot of content around the attacks like reviews etc. (see adoc files). I'm just wondering how we keep those in sync. It is not only the `message.properties` we need to translate.

File: src/main/java/org/owasp/webgoat/container/MvcConfiguration.java (hunk 1 above)
Comment (zubcevic) [reply src/main/java/org/owasp/webgoat/container/MvcConfiguration.java:205]: I will test if it is required, I don’t think so

File: src/main/java/org/owasp/webgoat/container/MvcConfiguration.java (hunk 2 above)
Comment (zubcevic) [reply src/main/java/org/owasp/webgoat/container/MvcConfiguration.java:230]: I will remove that

File: src/main/resources/lessons/webgoat_introduction/documentation/Introduction.adoc (hunk 3 above)
Comment (zubcevic) [reply src/main/resources/lessons/webgoat_introduction/documentation/Introduction.adoc]: Yes in the new commit I have support for using the lang as an attribute for use in adoc files, but I don’t think it makes things easier

File: src/main/resources/lessons/webgoat_introduction/documentation/Introduction_nl.adoc (hunk 4 above)
Comment (zubcevic) [reply src/main/resources/lessons/webgoat_introduction/documentation/Introduction_nl.adoc:18]: Yes and I’ll do the same for the german sentence

File: src/main/resources/lessons/webgoat_introduction/html/WebGoatIntroduction.html [hunk 5]
```diff
@@ -2,7 +2,17 @@
 <html xmlns:th="http://www.thymeleaf.org">
 
 <div class="lesson-page-wrapper">
-    <div class="adoc-content" th:replace="doc:lessons/webgoat_introduction/documentation/Introduction.adoc"></div>
+    <div th:switch="${#strings.toLowerCase(#locale.language)}">
```
[the remark this answers is not in the document]
Comment (zubcevic) [reply src/main/resources/lessons/webgoat_introduction/html/WebGoatIntroduction.html:5]: that would indeed be a bit more easy

**Comment by zubcevic** [comment]
Please review one more time. I think I tested all situations. Only the bootstrap based double dropdown is a bit strange in the behaviour. A kind of interference can occur if you click on both drop downs. 
I added a README_I18N.md for our convenience. Where it explains how a language is selected and what you need to change to support additional languages.

**Pull request merged**